sdks/agent-sdk/src/core/Agent.ts

```diff
@@ -175,13 +197,60 @@ export class Agent<ContentTypes> extends EventEmitter<
       this.#isListening = true;
       void this.emit("start");
 
-      const stream =
+      this.#conversationsStream = await this.#client.conversations.stream({
+        onValue: (conversation) => {
+          if (conversation instanceof Group) {
+            this.emit(
+              "group",
+              new ConversationContext<ContentTypes, Group<ContentTypes>>({
+                conversation,
+                client: this.#client,
+              }),
+            );
+          } else if (conversation instanceof Dm) {
+            this.emit(
+              "dm",
+              new ConversationContext<ContentTypes, Dm<ContentTypes>>({
+                conversation,
+                client: this.#client,
+              }),
+            );
+          }
+        },
+        onError: (error) => {
+          return this.#runErrorChain(error, null);
+        },
+        onFail: () => {
+          return this.#runErrorChain(
+            new Error("conversations.stream failed"),
+            null,
+          );
+        },
+      });
+
+      const messages =
```

Comment:
WDYT of doing the same thing here as with the conversation stream and have a `#messagesStream` field?

Reply:
Yeah, let's do it in the next PR.